rust/operator-binary/src/env_vars.rs

```diff
@@ -432,6 +437,59 @@ pub fn build_airflow_template_envs(
     transform_map_to_vec(env)
 }
 
+fn add_version_specific_env_vars(
+    airflow: &v1alpha1::AirflowCluster,
+    resolved_product_image: &ResolvedProductImage,
+    env: &mut BTreeMap<String, EnvVar>,
+) {
+    if resolved_product_image.product_version.starts_with("3.") {
+        env.extend(execution_server_env_vars(airflow));
+        env.insert(
+            AIRFLOW_CORE_AUTH_MANAGER.into(),
+            EnvVar {
+                name: AIRFLOW_CORE_AUTH_MANAGER.into(),
+                value: Some(
+                    "airflow.providers.fab.auth_manager.fab_auth_manager.FabAuthManager"
+                        .to_string(),
+                ),
+                ..Default::default()
+            },
+        );
+        env.insert(
+            AIRFLOW_API_AUTH_BACKENDS.into(),
+            EnvVar {
+                name: AIRFLOW_API_AUTH_BACKENDS.into(),
+                value: Some("airflow.api.auth.backend.session".into()),
+                ..Default::default()
+            },
+        );
+        // As of 3.x a JWT key is required.
+        // See https://airflow.apache.org/docs/apache-airflow/3.0.1/configurations-ref.html#jwt-secret
+        // This must be random, but must also be consistent across api-services.
+        // The key will be consistent for all clusters started by this
+        // operator instance. TODO: Make this cluster specific.
```

Comment:
Can you update the comment and mention https://github.com/stackabletech/airflow-operator/issues/639
This will make the TODO obsolete as well.